ksql-engine/src/main/java/io/confluent/ksql/KsqlEngine.java

```diff
@@ -304,26 +319,24 @@ public QueryMetadata getQueryExecutionPlan(final Query query, final KsqlConfig k
       // Parse and AST creation
       final KsqlParser ksqlParser = new KsqlParser();
 
-      final List<PreparedStatement> statements = ksqlParser.buildAst(
+      return ksqlParser.buildAst(
           queriesString,
           tempMetaStoreForParser,
-          stmt -> buildSingleQueryAst(
+          stmt -> postProcessSingleQueryAst(
               stmt.getStatement(),
               stmt.getStatementText(),
               tempMetaStore,
               tempMetaStoreForParser,
               convertStatementToQuery));
-
-      return statements
-          .stream()
-          .filter(Objects::nonNull)
-          .collect(Collectors.toList());
+    } catch (final KsqlException e) {
+      throw e;
     } catch (final Exception e) {
-      throw new ParseFailedException("Exception while processing statement: " + e.getMessage(), e);
+      throw new ParseFailedException(
+          "Exception while processing statements: " + e.getMessage(), queriesString, e);
     }
   }
 
-  private PreparedStatement buildSingleQueryAst(
+  private PreparedStatement postProcessSingleQueryAst(
```

Comment:
Hmm, `postProcess...` seems a bit wage, how about changing it to `prepare..` since it now returns `PreparedStatement`

Reply:
Please discuss with @rodesai, (who wanted this naming) and let me know the outcome.

Personally, I see this as post processing of the prepared statement, so I'm with @rodesai.